auth/data/src/main/java/com/nolawiworkineh/auth/data/NamePatternValidator.kt

```diff
@@ -0,0 +1,16 @@
+package com.nolawiworkineh.auth.data
+
+import com.nolawiworkineh.auth.domain.PatternValidator
+
+object NamePatternValidator : PatternValidator {
+    override fun matches(value: String): Boolean {
+        val name = value.trim()
+
+        if (name.length !in 4..50) {
+            return false
+        }
+        // The regex pattern checks for at least two words with letters only, separated by space
+        val namePattern = "^[A-Za-z]+(?:\\s[A-Za-z]+)+$".toRegex()
+        return namePattern.matches(value.trim())
```

Comment:
For simple patterns like a valid name, I'm always a friend of doing this manually since it lets you quickly know what makes a valid name. Something like this: 
```
val trimmed = value.trim()
val words = trimmed.split(" “)

val hasValidLength = trimmed.length in min..max // 4..50 in your case
val hasTwoWords = words.size >= 2
val hasOnlyLetters = words.all { word ->
  word.all { char -> 
    char.isLetter() 
    }
  }

return hasTwoWords && hasOnlyLetters && hasValidLength
```
That way, probably even my grandma could tell you what the requirements for a valid name are :D 
Regex makes sense if you can't easily do this manually, but has the cost of making code very unreadable.